src/main/java/bio/terra/common/exception/SamApiException.java

```diff
@@ -0,0 +1,86 @@
+package bio.terra.common.exception;
+
+import bio.terra.common.retry.SamRetry;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.http.HttpStatusCodes;
+import java.util.Collections;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsde.workbench.client.sam.ApiException;
+import org.broadinstitute.dsde.workbench.client.sam.model.ErrorReport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+
+/**
+ * Wrapper exception for non-200 responses from calls to Sam. Does not extend ErrorReportException
+ * so that SamApiException is a checked exception type. Can be converted to an ErrorReportException
+ * subtype using the toErrorReportException function.
+ * */
+public class SamApiException extends Exception {
+  private ApiException apiException;
+  private static final ObjectMapper objectMapper = new ObjectMapper();
+  private static Logger logger = LoggerFactory.getLogger(SamApiException.class);
+
+  static public SamApiException createSamApiException(ApiException apiException) {
```

Comment:
worth a few small unit tests on the conversion.

Reply:
Done